### ISSUE TITLE
Error when using spack create with backports.ssl_match_hostname

### ISSUE THREAD

**Issue by krafczyk** [LLNL/spack#2346]
When using `spack create` with backports.ssl_match_hostname, the following error occurs:

```
spack -d -v create https://pypi.io/packages/source/b/backports.ssl_match_hostname/backports.ssl_match_hostname-3.5.0.1.tar.gz
==> Error: Unknown namespace: backports
```

**Comment by adamjstewart** [comment]
I'll take a look at this. The problem is that I don't think Spack knows how to handle a package name with a period in the middle. In the meantime, you can specify the name you want as a quick workaround:
```
$ spack create --name ssl-match-hostname https://pypi.io/packages/source/b/backports.ssl_match_hostname/backports.ssl_match_hostname-3.5.0.1.tar.gz
```

**Comment by adamjstewart** [comment]
Fixed in #2351!
